modAionImpl/src/org/aion/zero/impl/cli/Cli.java

```diff
@@ -226,6 +226,12 @@ public ReturnType call(final String[] args, Cfg cfg) {
 
             // 5. options that can be influenced by the -d and -n arguments
 
+            if (options.getPort() != null) {
+                // update port in the config file
+                cfg.toXML(new String[] {"--p2p=" + ","+ options.getPort()}, configFile);
```

Comment:
I noticed you add a `,` to the string, then in CfgAion.java, where it is used, you split on the `,` character.  Since this parameter takes only a single argument, it would be more intuitive to not have the `,`.  And then just not do `String.split` at all in CfgAion.java.  For an example, see how CfgAion.java handles `--id=`

Reply:
this parameter takes two arguments, first one is <ip> second one is <port>, only <port> is getting overwritten in this case.